pkg/networking/cilium/reconciler/reconciler.go

```diff
@@ -260,3 +272,35 @@ func (r *Reconciler) installPreflight(ctx context.Context, client client.Client,
 
 	return nil
 }
+
+//nolint:gocyclo
+func (r *Reconciler) reconcileSpecialCases(ctx context.Context, c client.Client, yaml []byte) error {
+	objs, err := clientutil.YamlToClientObjects(yaml)
+	if err != nil {
+		return err
+	}
+
+	currentDS, currentDeployment, currentSvc, err := getObjects(ctx, c, cilium.DaemonSetName, cilium.DeploymentName)
+	if err != nil {
+		return err
+	}
+
+	for _, o := range objs {
+		if o.GetObjectKind().GroupVersionKind().GroupKind() == corev1.SchemeGroupVersion.WithKind("Service").GroupKind() && o.GetName() == cilium.ServiceName {
```

Comment:
should we store this in a var outside the method? `corev1.SchemeGroupVersion.WithKind("Service").GroupKind()`
to avoid recalculating